content/courses/advanced-ada/parts/data_types/records.rst

```diff
@@ -2124,19 +2124,506 @@ value set for the parent type :ada:`T`.
+function |mdash| instead of just hoping that an initialization function is
+called for this object at some point. Also, if further information is needed to
+initialize an object, we can add parameters to the :ada:`Init` function,
+thereby forcing the user to provide this information.
+
+For even more control over objects, we can use
+:ref:`limited types with unknown discriminants <Adv_Ada_Limited_Types_Unknown_Discriminants>`.
+
+
+.. _Adv_Ada_Unknown_Discriminants_Partial_Full_View:
+
+Partial and full view
+~~~~~~~~~~~~~~~~~~~~~
+
+As we've just seen, if we declare a type with an unknown discriminant part, we
+can only use it in the partial view. In the full view. we cannot use an unknown
+discriminant part, but have to use either no discriminants or known
+discriminants. For example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Data_Types.Records.Unknown_Discriminants.Partial_Full_View
+
+    package Unknown_Discriminants is
+
+       type Rec_No_Discr (<>) is private;
+
+       type Rec_Known_Discr (<>) is private;
+
+    private
+
+       type Rec_No_Discr is null record;
+
+       type Rec_Known_Discr
+         (L : Positive) is null record;
+
+    end Unknown_Discriminants;
+
+In this example, :ada:`Rec_No_Discr` has no discriminants in its full
+view, while :ada:`Rec_Known_Discr` has the discriminant :ada:`L`.
+
+In addition, the full view can be an (unconstrained) array type as well:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Data_Types.Records.Unknown_Discriminants.Partial_Full_View
+
+    package Unknown_Discriminants is
+
+       type Arr (<>) is private;
+
+    private
+
+       type Arr is
+         array (Positive range <>)
+           of Integer;
+
+    end Unknown_Discriminants;
+
+Here, the full view of :ada:`Arr` is an array type.
+
+.. admonition:: In the Ada Reference Manual
+
+   - :arm:`3.7 Discriminants <3-7>`
+
+
+.. _Adv_Ada_Unknown_Discriminants_Derived_Types:
+
+Derived types
+~~~~~~~~~~~~~
+
+As expected, we can derive from types with unknown discriminants. Consider the
+following package:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Data_Types.Records.Unknown_Discriminants.Derived_Type
+
+    package Unknown_Discriminants is
+
+       type Rec (<>) is private;
+
+    private
+
+       type Rec is null record;
+
+    end Unknown_Discriminants;
+
+We can then declare the :ada:`Derived_Rec` type:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Data_Types.Records.Unknown_Discriminants.Derived_Type
+
+    package Unknown_Discriminants.Children is
+
+       type Derived_Rec is
+         new Rec;
+
+    end Unknown_Discriminants.Children;
+
+Note that :ada:`Derived_Rec` has unknown discriminants, even though we're not
+explicitly using an unknown discriminant part (:ada:`(<>)`) in its declaration.
+(In fact, we're not allowed to use an unknown discriminant part in this case.)
+Therefore, declaring objects of this type directly isn't possible:
```

Comment:
I think you might want to emphasize that declaring objects of type Derived_Rec remains impossible, like the parent type Rec. My concern is that the reader might think that this is a new restriction, introduced by the derivation.
So maybe change
"directly isn't possible:"
to
"directly isn't possible, just like the parent type Rec:"

Reply:
Yes, that makes sense. Thanks for the suggestion!